### PR TITLE
feat: option for audio format

### PR DESCRIPTION
**Pull request by sapristi**
Add support for audio-format configuration.

Closes https://github.com/Spotifyd/spotifyd/issues/1081

### CODE REVIEW

File: src/main_loop.rs
Comment (eladyn) [inline src/main_loop.rs:169]: I guess, we can remove the comment now that you've hit such a case where it did not work and changed it accordingly :-)

**Review by eladyn — APPROVED** [review]
Looks good and works for me:

```console
$ target/debug/spotifyd --no-daemon --audio-format "S32"
Loading config from "/home/user/.config/spotifyd/spotifyd.conf"
No password specified. Checking password_cmd
No password_cmd specified
No proxy specified
Using software volume controller.
Connecting to AP "ap.spotify.com:443"
Authenticated as "user" !
Using Alsa sink with format: S32
Country: "DE"
...
```

Now we'll just have to wait for maintainers to come around and review this. :relaxed:

**Review by eladyn — COMMENTED** [review]
Just two additional nitpicks. Looking over `config.rs` I have some additional improvements in mind like [switching to `clap`](https://github.com/Spotifyd/spotifyd/issues/777) or [resurrecting this PR](https://github.com/Spotifyd/spotifyd/pull/800), but those are more general and definitely out-of-scope for this PR.

**Review by eladyn — DISMISSED** [review]

File: src/config.rs
Comment (eladyn) [reply src/config.rs:228]: > Also not much used to github PR, how the hell is there no comment thread ? Or am I missing something ?

You should in theory be able to reply inside this thread or did I misread your question?

> In the derive block, at least Deserialize is necessary. Otherwise I really don't have an opinion on this :)

Oh, right. We'll see what maintainers think, but I'd just leave it there for now.

**Comment by eladyn** [comment]
@sapristi With the recent PRs that have been merged, this PR now has merge conflicts. If you could resolve them, maybe we can get that into the next release?

**Comment by eladyn** [comment]
I took the liberty to take this PR and rebase it on the current master. So this could potentially also go into the next release.

**Review by slondr — APPROVED** [review]
awesome

**Review by slondr — APPROVED** [review]

**Pull request merged**
